tests/testing_utils.py

```diff
@@ -170,3 +174,31 @@ def with_production_testing(func):
     )
 
     return repository(hf_api(file_download(func)))
+
+
+def retry_endpoint(function, number_of_tries: int = 3, wait_time: int = 5):
+    """
+    Retries test if failure, waiting `wait_time`.
+    Should be added to any test hitting the `moon-staging` endpoint that is
+    downloading Repositories or uploading data
+
+    Args:
+        number_of_tries: Number of tries to attempt a passing test
+        wait_time: Time to wait in-between attempts in seconds
+    """
+
+    def decorator(*args, **kwargs):
+        retry_count = 1
+        while retry_count < number_of_tries:
+            try:
+                return function(*args, **kwargs)
+            except Exception:
```

Comment:
Ideally this would still catch the fact that it's a HTTP exception. If catching the `HTTPError` doesn't work, could you try and log the error so that we may see how to handle it?